expo/expo
# [image] Fix jest generate a long snapshot for the image component

# Why

Fixes jest generate a long snapshot for the image component.
Closes https://github.com/expo/expo/issues/26087. 

# How

The code uses the `JSON.stringify` function to serialize the react refs. However, most of them are removed from the snapshots by default because there is a logic that treats the `ref` prop differently. Unfortunately, this logic does not apply to our `nativeViewRef` prop, so it's serialized, resulting in a large snapshot. To change this default behavior, I added the `toJSON` function. Additionally, the logic was added to the `expo-modules-core` because we will need this in different packages too, such as `expo-video` or `expo-camera`.


# Test Plan

- run bare-expo ✅
- check if jest generates a smaller snapshot ✅
